src/eval.rs

```diff
@@ -215,27 +215,47 @@ impl Default for MiriConfig {
     }
 }
 
+struct MainThread<'tcx> {
+    entry_id: DefId,
+    entry_type: MiriEntryFnType,
+    args: Vec<String>,
```

Comment:
It should be enough to have just an `argc` and `argv`, both of type `ImmTy` here. You can do the setup of the arguments before calling the global ctors.

In fact you probably have to do that since the ctors may call shims that give access to the arguments and I assume that is supposed to work.